airbyte-cdk/python/airbyte_cdk/sources/source.py

```diff
@@ -66,7 +66,11 @@ def read_state(self, state_path: str) -> List[AirbyteStateMessage]:
                     parsed_state_messages.append(parsed_message)
                 return parsed_state_messages
             else:
-                # When the legacy JSON object format is received, always outputting an AirbyteStateMessage simplifies processing downstream
+                # Existing connectors that override read() might not be able to interpret the new state format. We temporarily
+                # send state in the old format for these connectors, but once all have been upgraded, this block can be removed
+                # vars(self.__class__) checks if the current class directly overrides the read() function
+                if "read" in vars(self.__class__):
+                    return state_obj
```

Comment:
Maybe we should log something warning the connector developer to do… something soon

Reply:
agreed, i think i'll jot this down to follow up and add this to the next checkin of per-stream. not sure the best way of surfacing it, maybe a logger message, but not really something we should be surfacing to customers viewing logs